src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -234,12 +234,40 @@ def show_webapp(cmd, resource_group_name, name, slot=None, app_instance=None):
     return webapp
 
 
-def update_webapp(instance, client_affinity_enabled=None, https_only=None):
+# for generic updater
+def get_app(cmd, resource_group_name, name, slot=None):
+    return _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'get', slot)
+
+
+def update_webapp(cmd, resource_group_name, name, slot=None,
+                  skip_dns_registration=None, skip_custom_domain_verification=None, force_dns_registration=None,
+                  ttl_in_seconds=None, client_affinity_enabled=None, https_only=None, **kwargs):
+    instance = kwargs['parameters']
+    if 'function' in instance.kind:
+        raise CLIError("please use 'az functionapp update' to update this function app")
     if client_affinity_enabled is not None:
         instance.client_affinity_enabled = client_affinity_enabled == 'true'
     if https_only is not None:
         instance.https_only = https_only == 'true'
```

Comment:
All of these `instance.<propery>` entries will remain part of the custom func.  It looks like that function will end here and return instance.

Reply:
I think I have done exactly you suggested, please double check.